recipe/meta.yaml

```diff
@@ -19,21 +19,21 @@ source:
     - h5py_win_setup_build.py.patch
 
 build:
-  number: 3
+  number: 4
 
 requirements:
   build:
     - python
     - numpy x.x
-    - hdf5 1.8.15*
+    - hdf5 1.8.17
```

Comment:
Seem to be missing a `*`, @pelson. This is a case where we don't want the `.` as that 4th version is included optionally.